metric.go

```diff
@@ -129,40 +118,27 @@ func makeBuckets(m *dto.Metric) map[string]string {
 	return result
 }
 
-func fetchMetricFamilies(url string, ch chan<- *dto.MetricFamily, certificate string, key string) {
+// FetchMetricFamilies does stuff
+func FetchMetricFamilies(url string, ch chan<- *dto.MetricFamily) {
 	defer close(ch)
-	var transport *http.Transport
-	if certificate != "" && key != "" {
-		cert, err := tls.LoadX509KeyPair(certificate, key)
-		if err != nil {
-			log.Fatal(err)
-		}
-		tlsConfig := &tls.Config{
-			Certificates: []tls.Certificate{cert},
-		}
-		tlsConfig.BuildNameToCertificate()
-		transport = &http.Transport{TLSClientConfig: tlsConfig}
-	} else {
-		transport = &http.Transport{}
-	}
-	client := &http.Client{Transport: transport}
-	decodeContent(client, url, ch)
-}
-
-func decodeContent(client *http.Client, url string, ch chan<- *dto.MetricFamily) {
 	req, err := http.NewRequest("GET", url, nil)
 	if err != nil {
 		log.Fatalf("creating GET request for URL %q failed: %s", url, err)
 	}
 	req.Header.Add("Accept", acceptHeader)
-	resp, err := client.Do(req)
+	resp, err := http.DefaultClient.Do(req)
 	if err != nil {
 		log.Fatalf("executing GET request for URL %q failed: %s", url, err)
 	}
 	defer resp.Body.Close()
 	if resp.StatusCode != http.StatusOK {
 		log.Fatalf("GET request for URL %q returned HTTP status %s", url, resp.Status)
 	}
+	ParseResponse(resp, ch)
+}
+
+// ParseResponse consumes an http.Response and pushes it to the MetricFamily channel
```

Comment:
End with period. Also, add the return behavior (see above, returns when all MetricFamilies have been sent to the channel).